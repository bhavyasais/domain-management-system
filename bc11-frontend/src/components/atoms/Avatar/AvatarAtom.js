import React from 'react';
import Avatar from '@material-ui/core/Avatar';

export default function AvatarAtom(props) {
    return (<Avatar src={props.src} className={props.className}/>);
}