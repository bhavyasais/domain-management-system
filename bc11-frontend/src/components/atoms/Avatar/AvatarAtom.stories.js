import React from "react";
import { storiesOf } from "@storybook/react";
import AvatarAtom from "./AvatarAtom";

storiesOf("Atoms/Avatar", module).add("Avatar", () => (
  <AvatarAtom
    src={
      "https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/screens/438d5cb7-5113-4c26-9a9f-109b56acf94c.png"
    }
  />
));
