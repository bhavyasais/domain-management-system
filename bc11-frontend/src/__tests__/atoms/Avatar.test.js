import Avatar from "../../components/atoms/Avatar/AvatarAtom";
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

describe("Avatar", () => {
  it("should render correctly", () => {
    let wrapper;
    wrapper = renderer.create(<Avatar />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders an image", () => {
    const wrapper = renderer.create(
      <Avatar src="https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/screens/438d5cb7-5113-4c26-9a9f-109b56acf94c.png" />
    );
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders given image", () => {
    const wrapper = mount(<Avatar src="i.png" />);
    expect(wrapper.props().src).toBe("i.png");
  });
});
