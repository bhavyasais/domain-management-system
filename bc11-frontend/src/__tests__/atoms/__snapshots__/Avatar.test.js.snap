// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders an image 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle"
>
  <img
    className="MuiAvatar-img"
    src="https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/screens/438d5cb7-5113-4c26-9a9f-109b56acf94c.png"
  />
</div>
`;

exports[`Avatar should render correctly 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiAvatar-fallback"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;
